Task4/Task4_2/RationalNumber.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task4_2
+{
+    internal sealed class RationalNumber : IComparable
+    {
+        public int Numerator { get; }
+
+        public byte Denominator { get; }
+
+        public RationalNumber(int numerator, byte denominator)
+        {
+            if (denominator == 0)
+            {
+                throw new Exception();
+            }
+
+            int gcd = GetGCD(numerator, denominator);
+
+            Numerator = numerator / gcd;
+            Denominator = (byte)(denominator / gcd);
+        }
+
+        public override bool Equals(object? obj)
+        {
+            if (obj is not RationalNumber || ((RationalNumber)obj).Denominator != this.Denominator || ((RationalNumber)obj).Numerator != this.Numerator)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public override string ToString()
+        {
+            string output = Numerator.ToString() + "/" + Denominator.ToString();
+            return output;
+        }
+
+        int GetGCD(int a, int b)
+        {
+            while (b != 0)
+            {
+                int temp = b;
+                b = a % b;
+                a = temp;
+            }
+            return a;
+        }
+
+        int GetLCM(int a, int b)
+        {
+            int gcd = GetGCD(a, b);
+            return Math.Abs(a * b) / gcd;
+        }
+
+        public int CompareTo(object? obj)
+        {
+            if (obj is RationalNumber)
+            {
```

Comment:
I see your idea to compare via cast to double, this is not a precise approach, and low performance. Try to find other decision

Reply:
Fixed